compiler/builtInMacros.go

```diff
@@ -0,0 +1,72 @@
+package compiler
+
+import "ddbt/fs"
+
+// All our built in Macros
+const builtInMacros = `
+{# This test checks that the value in column_name is always unique #}
+{% macro test_unique(model, column_name) %}
+	SELECT
+	{{ column_name }} AS value,
+	COUNT({{ column_name }}) AS count
+	
+	FROM {{ model }}
+	
+	GROUP BY {{ column_name }} 
+	
+	HAVING COUNT({{ column_name }}) > 1
+{% endmacro %}
+
+
+{# This test that the value is never null in column_name #}
+{% macro test_not_null(model, column_name) %}
+	SELECT
+	{{ column_name }} AS value
+	
+	FROM {{ model }}
+	
+	WHERE {{ column_name }} IS NULL
```

Comment:
Not 100% sure how tests work, but it might be useful to return the number of rows that have `NULL` values?

```
SELECT
COUNT(*) AS num_null_rows

FROM {{ model }}
WHERE {{ column_name }} IS NULL
```

Reply:
Ahh, I miss understood how DBT test work, I thought a failure was always when there was 1 or more rows present. Turns out the behaviour changes based on the type of test being executed;

> **schema tests (more common)**: applied in YAML, returns the number of records that do not pass an assertion — when this number is 0, all records pass, therefore, your test passes
**data tests**: specific queries that return 0 records